DataFormats/TrackerRecHit2D/interface/VectorHit2D.h

```diff
@@ -0,0 +1,32 @@
+#ifndef TrackerRecHit2D_VectorHit2D_h
+#define TrackerRecHit2D_VectorHit2D_h
+
+#include "DataFormats/GeometryVector/interface/LocalVector.h"
+#include "DataFormats/GeometryVector/interface/LocalPoint.h"
+#include "DataFormats/CLHEP/interface/AlgebraicObjects.h"
+#include "DataFormats/GeometrySurface/interface/LocalError.h"
+
+class VectorHit2D {
+public:
+  VectorHit2D() : thePosition(), theDirection(), theCovMatrix(), theChi2(), theDimension(2) {}
+  VectorHit2D(const LocalPoint& pos, const LocalVector& dir, const AlgebraicSymMatrix22& covMatrix, const double& Chi2)
+      : thePosition(pos), theDirection(dir), theCovMatrix(covMatrix), theChi2(Chi2), theDimension(2){};
+  virtual ~VectorHit2D(){};
+
+  LocalPoint localPosition() const { return thePosition; }
+  LocalVector localDirection() const { return theDirection; }
+  LocalError localDirectionError() const {
+    return LocalError(theCovMatrix[0][0], theCovMatrix[0][1], theCovMatrix[1][1]);
+  }
+  AlgebraicSymMatrix22 covMatrix() const { return theCovMatrix; }
+  double chi2() const { return theChi2; }
+  int dimension() const { return 2; }
```

Comment:
if this is defacto static constexpr, what use "theDimension" has?

Reply:
Good question. @ebrondol is there a reason to not return theDimension here? Is any dimension other than 2 even possible? If not we can remove theDimension.

Reply:
indeed, this class has a member `theDimension` which is hardcoded to 2 and then never accessed...

Reply:
Sounds reasonable. Still, the function `dimension` here should be modified to return `theDimension` instead of the hard-coded 2, right?

Reply:
Yes, as it is in the [VectorHit](https://github.com/cms-sw/cmssw/blob/8dfe7060aabdd9952538a33c1ee9cb912b94f8ec/DataFormats/TrackerRecHit2D/interface/VectorHit.h#L85) class